File: 프로그래머스/lv2/12914. 멀리 뛰기/멀리 뛰기.py
from itertools import permutations
def solution(n):
    f1, f2 = 1, 1
    if n < 2:
        return 1
    for i in range(n-1):
        f3 = (f1 + f2) % 1234567
        f1, f2 = f2, f3
    return f3

File: 프로그래머스/2/12914. 멀리 뛰기/멀리 뛰기.java
class Solution {
    public long solution(int n) {
        long f1 = 1, f2 = 1, f3 = 1;
        if(n < 2){
            return 1;
        }
        for(int i = 0; i < n - 1; i++){
            f3 = (f1 + f2) % 1234567;
            f1 = f2;
            f2 = f3;
        }
        return f3;
    }
}